WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java

```diff
@@ -246,6 +247,32 @@ public static ReaderTag getDefaultTag() {
         return defaultTag;
     }
 
+    public static @NonNull ReaderTag getDbOrInMemoryDefaultTag(
```

Comment:
Logic in switching from the concept of default tag to InMemoryDefault is not super clear to me, so I was wondering maybe you can explain what is going on here :) Word `memory` brings all kinds of questions :) Maybe we can simplify the naming in some way. Looking at the logic of this method, what I understand happens is - we are getting default tag, and if it was not existing and was just created, we turn it into `Following` tag.  If I my understanding is correct, would it be fair to call this method something like `getDefaultOrFollowingTag`?

Reply:
This one is not straightforward sorry 😊. 

Default Tag is actually the Following (at least currently). It is possible that that tag is not present in the db (for example when user has just switched app language or for network issues reasons). In those occasions there were logic in the `ReaderPostListFragment` that was trying to `ReaderTagTable.getFirstTag()` from the db, but also it is possible that `getFirstTag` returns null if tags table is empty (see #11206  for some more details and for a way to force the condition). 

In cases like this, we do not want to return a null tag but instead a `defaultTag` created in memory (not updating the db table) also populating the endpoint with the correct address (that is not the case when calling the standard `getDefaultTag` function. 

So the intention of the name `getDbOrInMemoryDefaultTag` was: get default tag from db or create it in memory. 

Tried to rename it like `getDefaultTagFromDbOrCreateInMemory` [here](6562ce83a81148058413e000e60371c01bc6d693) ; let me know if it seems a better fit or not, thanks 🙇‍♂️!